api/v1alpha1/llamastackdistribution_types.go

```diff
@@ -92,6 +95,38 @@ type UserConfigSpec struct {
 	ConfigMapNamespace string `json:"configMapNamespace,omitempty"`
 }
 
+// TLSConfig defines the TLS configuration for the llama-stack server
+type TLSConfig struct {
+	// CABundle defines the CA bundle configuration for custom certificates
+	// +optional
+	CABundle *CABundleConfig `json:"caBundle,omitempty"`
+}
+
+// CABundleConfig defines the CA bundle configuration for custom certificates
+// +kubebuilder:validation:XValidation:rule="!(has(self.configMapKey) && has(self.configMapKeys))",message="Only one of configMapKey or configMapKeys can be specified"
```

Comment:
This validation fails when I add `configMapKeys`

```
Danger alert:An error occurred
Error "Invalid value: "object": Only one of configMapKey or configMapKeys can be specified" for field "spec.server.tlsConfig.caBundle".
```

This is due to deafult value for `ConfigMapKey`. Remove the default -
```
// +kubebuilder:default:="ca-bundle.crt"
```

Alternatively we could just have only one field, `ConfigMapKeys`

Reply:
+1 to removing ConfigMapKey and just keeping ConfigMapKeys. I've made these changes in a local branch and will push once done addressing the recent comments.